src/components/AdministrationPage/AdministrationPage.tsx

```diff
@@ -38,18 +43,48 @@ export const AdministrationPage = () => {
   };
 
   const [announcementMessage, setAnnouncementMessage] = useState<string>('');
+  const [etaTime, setEtaTime] = useState<string>();
+  const [isEtaNa, setIsEtaNa] = useState<boolean>(false);
+  const [announcementTouched, setAnnouncementTouched] = useState<boolean>(false);
+  const [etaTouched, setEtaTouched] = useState<boolean>(false);
   const dataContainerAnnouncement = useDataContainer(
     ({ serviceData }: IService<string>) => genericSettingsService.setAnnouncementBanner(serviceData as string),
     {
       initLoadingState: false,
     }
   );
 
+  const validateForm = () => {
+    setAnnouncementTouched(true);
+    setEtaTouched(true);
+    return !invalidAnnouncement() && !invalidEtaTime();
+  };
+
+  const invalidAnnouncement = () => {
+    return isMaintenanceModeOn && announcementTouched && (!announcementMessage || announcementMessage === '');
+  };
+
+  const invalidEtaTime = () => {
+    return isMaintenanceModeOn && etaTouched && !isEtaNa && (!etaTime || etaTime === '');
+  };
+
   useEffect(() => {
     genericSettingsService
       .getAnnouncementBanner()
       .then((response: any) => {
-        setAnnouncementMessage(response.data.banner);
+        const rawAnnouncement: string = response.data.banner;
+        const rawAnnouncementSet: Array<string> = rawAnnouncement ? rawAnnouncement.split(', ETA: ') : [];
+        setAnnouncementMessage(rawAnnouncementSet[0]);
+        if (rawAnnouncementSet[1]) {
+          if (rawAnnouncementSet[1] !== 'N/A') {
```

Comment:
Maybe these endpoints could be used for getting / setting maintanance mode:
/generic-setting/in-maintenance-mode
/generic-setting/activate-maintenance-mode
/generic-setting/deactivate-maintenance-mode
(maybe this belongs to other jira)

Reply:
This was a part of Implement top bar announcement messages [NCL-7284](https://issues.redhat.com/browse/NCL-7284) or Integrate top bar announcement messages with WebSockets [NCL-7317](https://issues.redhat.com/browse/NCL-7317) because maintenance mode is regarded as a kind of announcement message. and I discussed with martin that we should leave it to you to fix it.

Reply:
Okay, I understand.